### DISCUSSION TITLE
PHOEBE2 installation in newer apple silicon macs

### DISCUSSION THREAD

**Discussion by tilakpawar** [phoebe-project/phoebe2#474] [Questions]
I was curious if any of the phoebe workshop (2021) attendees had the newer apple silicon (arm64) macs and if they could install and run phoebe2 successfully. I am having troubles doing the same. I guess the main problem is the compiler error because of the newer architecture but being no expert in this I am not sure if there are more problems, ones I cannot understand entirely. Any help would be appreciated :)
Thanks!!!

**Comment by kecnry** [comment]
I don't think any of the workshop attendees ran into this, but here are the steps @gecheline suggested, as long as you're willing to use conda:

* make sure you're running the terminal in emulation mode, here you can find instructions for that: https://medium.com/swlh/run-x86-terminal-apps-like-homebrew-on-your-new-m1-mac-73bdc9b0f343
* install miniconda (if you don't have conda installed already, you can find the bash script here: https://docs.conda.io/projects/continuumio-conda/en/latest/user-guide/install/macos.html)
* make a new environment to install phoebe: `conda create -n phoebe` and activate it `conda activate phoebe`
* install the 12.0.0 version of the clang compiler with `conda install -c conda-forge clang=12.0.0`
* install pip: `conda install pip`
* install numpy and phoebe with pip

If you get a RuntimeError mentioning forking child processes, use `phoebe.multprocessing_off()` or permanently disable by setting the `PHOEBE_MULTIPROC_NPROCS=0` environment variable.

Let us know if this works for you or any edits you'd suggest and I'll update the [installation instructions page of the website](http://phoebe-project.org/install/latest/mac/auto).  Thanks!

**Reply by tilakpawar** [reply]
**Thanks a lot for such a quick and comprehensive reply! I had anaconda installed in the system but I installed miniconda from scratch, just to be sure and could successfully install and run phoebe. I did not seem to run into the RuntimeError you mentioned.
A small note, while installing 'ellc' using pip I encountered the following error:**

WARNING: Discarding https://files.pythonhosted.org/packages/6f/12/bfe4450bafd950790758533136c8d4b75266df50883083ca05f6bb1d780e/ellc-1.8.4.tar.gz#sha256=a8970c7b03f0aea148458d34b8a006379e130f3a5c4ce1aaae74c41f81b86281 (from https://pypi.org/simple/ellc/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached ellc-1.8.3.tar.gz (13.7 MB)
    ERROR: Command errored out with exit status 1:
     command: /opt/homebrew/opt/python@3.9/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/ks/jh9sq_45137fbd82mvsrvwpm0000gn/T/pip-install-cxczrw8v/ellc_076981923b9644d6ac2908ea21bbb848/setup.py'"'"'; __file__='"'"'/private/var/folders/ks/jh9sq_45137fbd82mvsrvwpm0000gn/T/pip-install-cxczrw8v/ellc_076981923b9644d6ac2908ea21bbb848/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/ks/jh9sq_45137fbd82mvsrvwpm0000gn/T/pip-pip-egg-info-ugj3p3u9
         cwd: /private/var/folders/ks/jh9sq_45137fbd82mvsrvwpm0000gn/T/pip-install-cxczrw8v/ellc_076981923b9644d6ac2908ea21bbb848/
    Complete output (45 lines):
    Traceback (most recent call last):
      File "/opt/homebrew/lib/python3.9/site-packages/numpy/core/__init__.py", line 22, in <module>
        from . import multiarray
      File "/opt/homebrew/lib/python3.9/site-packages/numpy/core/multiarray.py", line 12, in <module>
        from . import overrides
      File "/opt/homebrew/lib/python3.9/site-packages/numpy/core/overrides.py", line 7, in <module>
        from numpy.core._multiarray_umath import (
    ImportError: dlopen(/opt/homebrew/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so, 2): no suitable image found.  Did find:
        /opt/homebrew/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so: mach-o, but wrong architecture
        /opt/homebrew/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so: mach-o, but wrong architecture
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/ks/jh9sq_45137fbd82mvsrvwpm0000gn/T/pip-install-cxczrw8v/ellc_076981923b9644d6ac2908ea21bbb848/setup.py", line 34, in <module>
        from numpy.distutils.core import setup
      File "/opt/homebrew/lib/python3.9/site-packages/numpy/__init__.py", line 145, in <module>
        from . import core
      File "/opt/homebrew/lib/python3.9/site-packages/numpy/core/__init__.py", line 48, in <module>
        raise ImportError(msg)
    ImportError:
    
    IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!
    
    Importing the numpy C-extensions failed. This error can happen for
    many reasons, often due to issues with your setup or how NumPy was
    installed.
    
    We have compiled some common reasons and troubleshooting tips at:
    
        https://numpy.org/devdocs/user/troubleshooting-importerror.html
    
    Please note and check the following:
    
      * The Python version is: Python3.9 from "/opt/homebrew/opt/python@3.9/bin/python3.9"
      * The NumPy version is: "1.20.3"
    
    and make sure that they are the versions you expect.
    Please carefully study the documentation linked above for further help.
    
    Original error was: dlopen(/opt/homebrew/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so, 2): no suitable image found.  Did find:
        /opt/homebrew/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so: mach-o, but wrong architecture
        /opt/homebrew/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so: mach-o, but wrong architecture

**The error could be resolved by installing 'ellc' via conda using conda-forge. I will keep you posted if anything else relating to the compatibility issue pops up. Thanks again! :)**

**Reply by gecheline** [reply]
I ran into the same issue with ellc through pip, and I think it has to do with the fortran compilers it uses under the hood (I tried to install the equivalent of what it was pulling from homebrew in conda with no luck). There's also the architecture mismatch (homebrew already offers arm64 architecture, while conda doesn't). Thanks for providing the solution using conda-forge, I hadn't thought to look if ellc is available through it!

**Comment by amiszuda** [comment]
Trying the above on M2 gives:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/amiszuda/miniconda3/envs/phoebe_tmp/lib/python3.11/site-packages/phoebe/__init__.py", line 377, in <module>
    from .atmospheres.passbands import install_passband, uninstall_passband, uninstall_all_passbands, download_passband, list_passband_online_history, update_passband_available, update_passband, update_all_passbands, list_all_update_passbands_available, list_online_passbands, list_installed_passbands, list_passbands, list_passband_directories, get_passband
  File "/Users/amiszuda/miniconda3/envs/phoebe_tmp/lib/python3.11/site-packages/phoebe/atmospheres/passbands.py", line 22, in <module>
    import libphoebe
ImportError: dlopen(/Users/amiszuda/miniconda3/envs/phoebe_tmp/lib/python3.11/site-packages/libphoebe.cpython-311-darwin.so, 0x0002): tried: '/Users/amiszuda/miniconda3/envs/phoebe_tmp/lib/python3.11/site-packages/libphoebe.cpython-311-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/amiszuda/miniconda3/envs/phoebe_tmp/lib/python3.11/site-packages/libphoebe.cpython-311-darwin.so' (no such file), '/Users/amiszuda/miniconda3/envs/phoebe_tmp/lib/python3.11/site-packages/libphoebe.cpython-311-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
```
